jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/MovedContextHandler.java

```diff
@@ -102,8 +102,8 @@ public void process(Request request, Response response, Callback callback) throw
                 return;
 
             String path = _newContextURL;
-            if (!_discardPathInfo && request.getPathInContext() != null)
-                path = URIUtil.addPaths(path, request.getPathInContext());
+            if (!_discardPathInfo && Request.getPathInContext(request) != null)
+                path = URIUtil.addPaths(path, Request.getPathInContext(request));
```

Comment:
Store `Request.getPathInContext(request)` in a local variable to avoid calling it twice.